lib/floe/workflow/intrinsic_function.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+module Floe
+  class Workflow
+    class IntrinsicFunction
+      INTRINSIC_FUNCTION_REGEX = /^(?<module>\w+)\.(?<function>\w+)\((?<args>.*)\)$/.freeze
+
+      class << self
+        def detect_class(payload)
+          function_module_name, function_name =
+            payload.match(INTRINSIC_FUNCTION_REGEX)
+                   .named_captures
+                   .values_at("module", "function")
+
+          begin
+            function_module = Floe::Workflow::IntrinsicFunctions.const_get(function_module_name)
+            function_module.const_get(function_name)
+          rescue NameError
+            raise NotImplementedError, "#{function_module_name}.#{function_name} is not implemented"
+          end
+        end
+
+        def value(payload, context, input = {})
+          new(payload).value(context, input)
+        end
+
+        def new(*args)
+          if self == Floe::Workflow::IntrinsicFunction
+            detect_class(*args).new(*args)
+          else
+            super
+          end
+        end
```

Comment:
Extremely, minor, but can we swap the order of these 2 methods?  I feel like ".new" is "more important" than ".value" and also I feel it should live next to detect_class.  Also, I'm wondering if detect_class should be marked private and perhaps listed _after_ ".new" (giving a final method ordering of "new", "detect_class", "value")